### PR TITLE
Add plugin: Navigable File Tree

### PR DESCRIPTION
**Pull request by Jaschenn**
add navigable-file-tree plug

# I am submitting a new Community Plugin

## Repo URL

[Link to my plugin](https://github.com/Jaschenn/navigable-file-tree)

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by Jaschenn** [comment]
remove obsidian keyword from repo description

**Comment by Jaschenn** [comment]
i found some bug this morning, and i have already fix it. (also update v1.0.0 release file). 
do i need some extra flow to handle this pr  or just wait human review ?

**Comment by joethei** [comment]
To make the manual review process and future plugin maintenance more efficient, we recommend that you move all source code files into a `src` directory.

["description": "A file tree plugin with navigation bar for Obsidian",](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/manifest.json#L6)
Avoid including sentences like "This is a plugin for Obsidian that does ..." in your description, it should be a given since this is only shown when browsing plugins through the app or on our website.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

["main": "main.js"](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/manifest.json#L9)
This is not a valid part of the manifest, remove it.

['Navigable File Tree'](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/main.ts#L24), [return 'Navigable File Tree';](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L44), [tooltip: 'Expand/Collapse All',](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L425), [tooltip: 'Create New',](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L432), [item.setTitle('New Note')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L547), [item.setTitle('New Folder')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L559), [item.setTitle('Sort by Name')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L570), [item.setTitle('Sort by Modified')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L578), [item.setTitle('Open in New Tab')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L612), [item.setTitle('Remove from Navigation')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L622), [item.setTitle('Pin to Navigation')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1102), [item.setTitle('Open in New Tab')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1157), [item.setTitle('Pin to Navigation')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1166), [item.setTitle('New Note')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1195), [item.setTitle('New Folder')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1207), [item.setTitle('Pin to Navigation')](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1215)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[leaves.forEach(leaf => leaf.detach());](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/main.ts#L64)
Do not detach leaves with your custom view here, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>
Instead, only create a new leaf, if there is not one already, if there is one, focus on it.

[let dropTimeout: NodeJS.Timeout;](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L947)
Use `number` as the type here, and use the `window.setTimeout` / `window.clearTimeout` function where this is called.

[const leaf = this.app.workspace.splitActiveLeaf();](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/FileTreeView.ts#L1160)
That function has been deprecated for years, use the documented replacement.

[.tree-item {](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/styles.css#L207-L212), [.collapse-icon {](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/styles.css#L243-L253), [.tree-item:hover > .file-title,](https://github.com/Jaschenn/navigable-file-tree/blob/ff545bad252f72bc700d19172fb282dddb6544a1/styles.css#L267-L284)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.
